docs/sphinx/developers/logging.txt

```diff
@@ -0,0 +1,51 @@
+Logging
+=======
+
+Logging is the process of writing log messages during execution.
```

Comment:
A single introductory sentence was certainly either too much or not enough. I will remove it as I don't have the capacity to write an authoritative introduction about logging.